tests.cpp

```diff
@@ -0,0 +1,53 @@
+#define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this
+                          // in one cpp file
+#include "HybridCar.hpp"
+#include "catch.hpp"
+//#include "WrongGearException.hpp"
+
+// Electric Car is best implemented, so I will check only electric Car
+
+SCENARIO("Checking if Electric functions works", "[ElectricCar]") {
+
+  GIVEN("ElectricCar, car raw_ptr and Shop") {
+    ElectricCar Tesla(std::make_unique<ElectricEngine>(600, 4000));
+    std::vector<std::unique_ptr<ElectricEngine>> OldEngineShop;
+
+    REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() == 4000);
+    REQUIRE(Tesla.getElectricEngineData()->getPower() >= 600);
```

Comment:
If you reuse some data it's better to assign it to a variable. It's a Demeter rule.

```suggestion
    auto electricEngine = Tesla.getElectricEngineData();
    REQUIRE(electricEngine->getBatteryCapacity() == 4000);
    REQUIRE(electricEngine->getPower() >= 600);
```

The same for all other places